beacon-chain/endtoend/evaluators.go

```diff
@@ -0,0 +1,144 @@
+package endtoend
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	ptypes "github.com/gogo/protobuf/types"
+	"github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// policy defines an function that returns a bool
+// for whether the evaluator it is assigned to runs.
+type policy func(currentEpoch uint64) bool
+
+// evaluation defines a function that takes in BeaconChainClient
+// and uses it to monitor the chain, to determine where it fulfills the conditions required.
+type evaluation func(client eth.BeaconChainClient) error
+
+// evaluator defines a struct for executing a given evaluation
+// on a running E2E test given that the policy returns true.
+type evaluator struct {
+	name       string
+	policy     policy
+	evaluation evaluation
+}
+
+// runEvaluators takes in testing.T, BeaconChainClient and an array of evaluators.
```

Comment:
I would probably rename `testing.T` and `BeaconChainClient` to somewhere else

Reply:
No need to explain in comments what the arguments to the func are unless super unclear, just mention that it uses the client to check the state and run each in a subset